packages/webapp/pages/onboarding.tsx

```diff
@@ -1,591 +1,366 @@
-
-  const showOnboardingPage = !isAuthenticating && !shouldVerify && isIntro;
-  const showBackground =
-    showOnboardingPage || isExperimental.reorder.registration;
-  const showGenerigLoader =
-    isAuthenticating && isAuthLoading && isIntro && !isOnboardingReady;
-
-  const layout = useMemo(
-    () => ({
-      hasCta: onboardingStepsWithCTA.includes(activeScreen),
-      hasFooterLinks: onboardingStepsWithFooter.includes(activeScreen),
-    }),
-    [activeScreen],
+    [setAuth],
   );
 
-  // Ensure dark mode for authentication screen, since we only have a darkmode background image
-  useEffect(() => {
-    if (!user) {
-      applyThemeMode(ThemeMode.Dark);
-    }
-    return () => {
-      applyThemeMode();
-    };
-  }, [applyThemeMode, user]);
-
+  // Update the Jotai auth state based on the login state and anonymous user
   useEffect(() => {
-    if (
-      !isPageReady ||
-      isLogged.current ||
-      !isOnboardingReady ||
-      !user?.infoConfirmed
-    ) {
+    const email = loginState?.formValues?.email || anonymous?.email;
+    const shouldVerify = anonymous?.shouldVerify;
+    const isLogin = loginState?.isLogin;
+    const isRequiredAuth = !(isLoggedIn && !shouldVerify);
+    const hasLoginState =
+      !!loginState?.formValues?.email || loginState?.isLogin;
+    const wasLoggedInBefore = !!storage.getItem(SIGNIN_METHOD_KEY);
+
+    if (!isAuthReady) {
       return;
     }
 
-    isLogged.current = true;
-
-    if (!hasCompletedEditTags) {
-      if (getFeatureValue(featureInteractiveFeed) && isLaptop) {
-        setActiveScreen(OnboardingStep.InteractiveFeed);
-      } else {
-        setActiveScreen(OnboardingStep.EditTag);
-      }
-      return;
-    }
-
-    if (!hasCompletedContentTypes) {
-      setActiveScreen(OnboardingStep.ContentTypes);
-      return;
-    }
-
-    if (isIntro) {
-      const params = new URLSearchParams(window.location.search);
-      const afterAuth = params.get(AFTER_AUTH_PARAM);
-      params.delete(AFTER_AUTH_PARAM);
-      router.replace(getPathnameWithQuery(afterAuth || webappUrl, params));
-    }
-
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+    updateAuth({
+      defaultDisplay: getDefaultDisplay({
+        isLogin,
+        shouldVerify,
+        action,
+        wasLoggedInBefore,
+      }),
+      email,
+      isAuthenticating:
+        isRequiredAuth &&
+        (!!action || shouldVerify || wasLoggedInBefore || hasLoginState),
+      isLoading: !isAuthReady,
+      isLoginFlow: loginState?.isLogin || action === OnboardingActions.Login,
+    });
   }, [
-    isIntro,
-    isPageReady,
-    user,
-    isOnboardingReady,
-    hasCompletedEditTags,
-    hasCompletedContentTypes,
-    activeScreen,
-    getFeatureValue,
+    action,
+    anonymous?.email,
+    anonymous?.shouldVerify,
+    isAuthReady,
+    isLoggedIn,
+    loginState?.formValues?.email,
+    loginState?.isLogin,
+    updateAuth,
```

Comment:
This useEffect feels pretty dangerous. Why don't we run this call whenever we need to and not based from a side effect?